java/runtime/src/main/java/org/ray/runtime/runner/worker/DefaultWorker.java

```diff
@@ -15,6 +16,7 @@
   public static void main(String[] args) {
     try {
       System.setProperty("ray.worker.mode", "WORKER");
+      Thread.setDefaultUncaughtExceptionHandler(new WorkerUncaughtExceptionHandler());
```

Comment:
I don't think It's a good idea to catch all exceptions. Why not simply redirect `stderr` to the log file?

I guess the process will not crash even if there's a runtime exception in main thread.

Reply:
The main thread will catch all the exceptions. For exceptions in other threads, user can catch the exception without triggering this hander. This handler will only be triggered when the exception is not caught by user. It may not be a good idea to suppress this exception. Maybe we need to re-throw this exception? After re-throwing, the error message may still appear in `raylet.err`. If we redirect the stderr, can this message be redirected to the log4j log file? Can threads write the same file at the same time?

Reply:
1) Using `LOGGER.error` can ensure consistent log format. 
2) This doesn't impact main thread, only the background threads.
3) If we don't set this, the process won't crash, because we catch all exceptions in Worker.java

Reply:
@raulchen One question, after using `setDefaultUncaughtExceptionHandler `, will `Code snippet 2` be executed in the following code example in another thread? If it is yes, the loggic maybe strange. However, I think it probably won't be executed.
```java
Code snippet 1
throw new Exception();
Code snippet 2
```

Reply:
I guess it won't execute.

Reply:
I tried the following code:
```java
@Test
  public void testMultithreadException() {
    Thread.setDefaultUncaughtExceptionHandler(new WorkerUncaughtExceptionHandler());
    Thread thread = new Thread(() -> {
      System.out.println("Before exception.");
      int a = 1/0;
      System.out.println("After exception.");
    });
    thread.start();
    try {
      thread.join();
    } catch (Exception e) {
      System.out.println(e);
    }
  }
```
The output is:
```
2019-05-16 17:31:37 INFO RayNativeRuntime [main]: RayNativeRuntime started with store /tmp/ray/sockets/object_store, raylet /tmp/ray/sockets/raylet
Before exception.
2019-05-16 17:31:37 ERROR WorkerUncaughtExceptionHandler [Thread-4]: Uncaught exception in thread Thread[Thread-4,5,main]: {}
java.lang.ArithmeticException: / by zero
	at org.ray.api.test.MultiThreadingTest.lambda$testMultithreadException$0(MultiThreadingTest.java:53)
	at java.lang.Thread.run(Thread.java:748)

===============================================
Default Suite
Total tests run: 1, Failures: 0, Skips: 0
===============================================
```
Therefore, I think the exception logic won't be break. It is fine to use this handler.